parastell/utils.py

```diff
@@ -56,124 +58,63 @@ def expand_ang_list(ang_list, num_ang):
-#             and geometry information (bool).
-#         'legacy_faceting': choose legacy or native faceting for DAGMC
-#             export (bool).
-#         'faceting_tolerance': maximum distance a facet may be from
-#             surface of CAD representation for DAGMC export (double).
-#         'length_tolerance': maximum length of facet edge for DAGMC
-#             export (double).
-#         'normal_tolerance': maximum change in angle between normal
-#             vector of adjacent facets (double).
-#         'anisotropic_ratio': controls edge length ratio of elements
-#             (double).
-#         'deviation_angle': controls deviation angle of facet from
-#             surface (i.e., lesser deviation angle results in more
-#             elements in areas with greater curvature) (double).
-#         'filename': name of DAGMC output file, excluding '.h5m'
-#             extension (str, defaults to 'dagmc').
-#         'export_dir': directory to which to export DAGMC output file
-#             (str, defaults to empty string).
-#             }
-dagmc_export_def = {
-    'skip_imprint': False,
-    'legacy_faceting': True,
-    'faceting_tolerance': None,
-    'length_tolerance': None,
-    'normal_tolerance': None,
-    'anisotropic_ratio': 100,
-    'deviation_angle': 5,
-    'filename': 'dagmc',
-    'export_dir': ''
-}
+
+def construct_kwargs_from_dict(
+    dict, allowed_kwargs, all_kwargs, fn_name=None, logger=None
+):
+    """Constructs a dictionary of keyword arguments with corresponding values
+    for a class method from an input dictionary based on a list of allowable
+    keyword argument argument names. Conditionally raises an exception if the
+    user flags that the supplied list of allowable arguments should represent
+    all keys present in the input dictionary of arguments.
+
+    Arguments:
+        dict (dict): dictionary of arguments and corresponding values.
+        allowed_kwargs (list of str): list of allowed keyword argument names.
+        all_kwargs (bool): flag to indicate whether 'allowed_kwargs' should
+            represent all keys present in 'dict'.
+        fn_name (str): name of class method (optional, defaults to None). If
+            'all_kwargs' is True, a method name should be supplied.
+        logger (object): logger object (optional, defaults to None). If
+            'all_kwargs' is True, a logger object should be supplied.
+
+    Returns:
+        kwarg_dict (dict): dictionary of keyword arguments and values.
+    """
+    kwarg_dict = {}
+    for name, value in dict.items():
+        if name in allowed_kwargs:
+            kwarg_dict.update({name: value})
+        elif all_kwargs:
+            e = ValueError(
+                f'{name} is not a supported keyword argument of '
+                f'"{fn_name}"'
+            )
+            logger.error(e.args[0])
+            raise e
+
+    return kwarg_dict
+
+
+def set_kwarg_attrs(
+    class_obj, kwargs, allowed_kwargs, logger
```

Comment:
is it reasonable to assume that the `logger` should always be `class_obj.logger` and not need to pass in an extra variable?